checkov/sast/consts.py

```diff
@@ -12,8 +15,104 @@ def list(cls) -> List[Any]:
     JAVASCRIPT = 'javascript'
 
 
+class SemgrepAttribute(str, Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    ID = 'id'
+    MESSAGE = 'message'
+    SEVERITY = 'severity'
+    CWE = 'cwe'
+    OWASP = 'owasp'
+    LANGUAGES = 'languages'
+    PATTERNS = 'patterns'
+    PATTERN = 'pattern'
+    PATTERN_EITHER = 'pattern-either'
+    PATTERN_INSIDE = 'pattern-inside'
+    PATTERN_NOT_INSIDE = 'pattern-not-inside'
+    PATTERN_NOT = 'pattern-not'
+    PATTERN_REGEX = 'pattern-regex'
+    PATTERN_NOT_REGEX = 'pattern-not-regex'
+    METAVARIABLE = 'metavariable'
+    METAVARIABLE_PATTERN = 'metavariable-pattern'
+    METAVARIABLE_REGEX = 'metavariable-regex'
+    METAVARIABLE_COMPARISON = 'metavariable-comparison'
+    COMPARISON = 'comparison'
+    REGEX = 'regex'
+
+
+class BqlConditionType(str, Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    PATTERN = "pattern"
+    VARIABLE = "variable"
+    FILTER = 'filter'
+    OR = 'or'
+    AND = 'and'
+
+
+PATTERN_OPERATOR_TO_SEMGREP_ATTR = {
+    Operators.EQUALS: SemgrepAttribute.PATTERN.value,
+    Operators.NOT_EQUALS: SemgrepAttribute.PATTERN_NOT.value,
+    Operators.REGEX_MATCH: SemgrepAttribute.PATTERN_REGEX.value,
+    Operators.NOT_REGEX_MATCH: SemgrepAttribute.PATTERN_NOT_REGEX.value
+}
+
+VARIABLE_OPERATOR_TO_SEMGREP_ATTR = {
+    Operators.REGEX_MATCH: SemgrepAttribute.METAVARIABLE_REGEX.value,
+    Operators.PATTERN_MATCH: SemgrepAttribute.METAVARIABLE_PATTERN.value,
+    Operators.EQUALS: SemgrepAttribute.METAVARIABLE_COMPARISON.value,
+    Operators.NOT_EQUALS: SemgrepAttribute.METAVARIABLE_COMPARISON.value,
+    Operators.GREATER_THAN: SemgrepAttribute.METAVARIABLE_COMPARISON.value,
+    Operators.GREATER_THAN_OR_EQUAL: SemgrepAttribute.METAVARIABLE_COMPARISON.value,
+    Operators.LESS_THAN: SemgrepAttribute.METAVARIABLE_COMPARISON.value,
+    Operators.LESS_THAN_OR_EQUAL: SemgrepAttribute.METAVARIABLE_COMPARISON.value
+}
+
+FILTER_OPERATOR_TO_SEMGREP_ATTR = {
+    Operators.WITHIN: SemgrepAttribute.PATTERN_INSIDE.value,
+    Operators.NOT_WITHIN: SemgrepAttribute.PATTERN_NOT_INSIDE.value,
+}
+
 SUPPORT_FILE_EXT = {
     SastLanguages.PYTHON: ['py'],
     SastLanguages.JAVA: ['java'],
     SastLanguages.JAVASCRIPT: ['js']
 }
+
+FILE_EXT_TO_SAST_LANG = {
+    'py': SastLanguages.PYTHON,
+    'java': SastLanguages.JAVA,
+    'js': SastLanguages.JAVASCRIPT
```

Comment:
do we also want to add `'ts': SastLanguages.JAVASCRIPT` or will it be an exclusive language?

Reply:
I'm not sure how ts scanning works with semgrep and if it gets under javascript or not, but Iet's leave it as an open question and possible modification for when we implement the js policies and get the answer